repositories/src/main/scala/cromwell/pipeline/datastorage/dto/ProjectSearch.scala

```diff
@@ -0,0 +1,68 @@
+package cromwell.pipeline.datastorage.dto
+
+import play.api.libs.json._
+
+sealed trait ProjectSearchFilter
+object ProjectSearchFilter {
+  implicit lazy val projectSearchFilterReads: Reads[ProjectSearchFilter] = Reads[ProjectSearchFilter] { json =>
+    (json \ "type").validate[String].flatMap {
+      case "all"            => json.validate[AllFilter.type]
+      case "name"           => json.validate[ByNameFilter]
+      case "files"          => json.validate[ByFilesFilter]
+      case "configurations" => json.validate[ByConfigFilter]
+      case "OR"             => json.validate[ComplexOrFilter]
+      case "AND"            => json.validate[ComplexAndFilter]
+    }
+  }
+}
+
+case object AllFilter extends ProjectSearchFilter {
+  implicit lazy val allReads: Reads[AllFilter.type] = Json.reads[AllFilter.type]
+}
+
+final case class ByNameFilter(mode: NameSearchMode, value: String) extends ProjectSearchFilter
+object ByNameFilter {
+  implicit lazy val byNameReads: Reads[ByNameFilter] = Json.reads[ByNameFilter]
+}
+
+sealed trait NameSearchMode
+case object FullMatch extends NameSearchMode
+case object RegexpMatch extends NameSearchMode
+
+object NameSearchMode {
+  implicit lazy val nameSearchModeReads: Reads[NameSearchMode] = {
+    implicitly[Reads[String]].map(NameSearchMode.fromString)
+  }
+
+  def fromString(s: String): NameSearchMode = s match {
+    case "full_match"   => FullMatch
+    case "regexp_match" => RegexpMatch
+  }
+}
+
+final case class ByConfigFilter(mode: String, value: Boolean) extends ProjectSearchFilter
+object ByConfigFilter {
+  implicit lazy val byConfigFilterReads: Reads[ByConfigFilter] = Json.reads[ByConfigFilter]
+}
+
+final case class ByFilesFilter(mode: String, value: Boolean) extends ProjectSearchFilter
+object ByFilesFilter {
+  implicit lazy val byFilesFilterReads: Reads[ByFilesFilter] = Json.reads[ByFilesFilter]
+}
+
+final case class ComplexOrFilter(left: ProjectSearchFilter, right: ProjectSearchFilter) extends ProjectSearchFilter
```

Comment:
Can't we just name them as `Or` and `And` instead of `ComplexOrFilter` and `ComplexAndFilter` ?

Reply:
Fixed